project-koku/koku
# [COST-4109] Start subs processing at midnight day before source creation

## Jira Ticket

[COST-4109](https://issues.redhat.com/browse/COST-4109)

## Description

This change will start gathering subs data from midnight the day prior to source creation rather than as far back as we have data for.

## Testing

1. Checkout Branch
2. Set `ENABLE_SUBS_DEBUG` to `True` in your .env
3. Restart Koku with subs `make docker-up-min-with-subs`
4. Create and load the test customer data.
5. Check the minio bucket for subs data under a path like: 
```
koku-bucket/org1234567/AWS/source=a3d4f4a3-cb3a-44ed-839d-5bc7015548d7/date=2023-09-11/subs_2099e908-c7d4-4ddd-8f6f-42c5afcc76d8_0.csv
```
6. Verify the information in the subs CSV starts at midnight the day prior to source creation by looking at the `lineitem_usagestartdate` column.

Advanced Test:
1. Follow steps 1-3 above
2. Create the test customer `make create-test-customer` then update the record in the DB that corresponds to the AWS source for the test customer. Update the timestamp to a different day in the past like so:
```
UPDATE api_provider SET created_timestamp='2023-09-08 12:50:08.472567+00' WHERE uuid='b50cc4e1-f5e3-4149-9261-10217ee7f98b';
```
3. Load the test customer data
4. Check the minio bucket and verify the information in the subs csv starts at midnight the day prior to the new time you set for the provider creation.

## Notes

...
